spacy_loggers/cupy.py

```diff
@@ -0,0 +1,71 @@
+"""
+A logger that queries CuPy metrics and passes that information to downstream loggers.
+"""
+from typing import Dict, Any, Optional, IO
+import sys
+
+from spacy import Language
+from thinc.backends import context_pools
+from thinc.util import has_cupy_gpu
+from .util import LoggerT
+
+
+def cupy_logger_v1(
+    prefix: str = "cupy",
+) -> LoggerT:
+    """Creates a logger that queries CuPy metrics and passes that information to downstream loggers.
+
+    Args:
+        prefix (str):
+            All metric names are prefixed with this string using dot notation, e.g: `<prefix>.<metric>`.
+
+    Returns:
+        LoggerT: Logger instance.
+    """
+    try:
+        import cupy
+
+        if not has_cupy_gpu:
```

Comment:
Why isn't the `has_cupy_gpu` check outside the `try` block? It's not like this will throw an `ImportError`, right?